src/App.js

```diff
@@ -27,45 +26,77 @@ const generateSquares = () => {
 
 const App = () => {
 
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState('X');
 
   // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
 
+  const updateSquare = (squareToUpdate) => {
+    
+    const row = Math.floor(squareToUpdate.id/3);
+    const column = squareToUpdate.id%3;
+    const newBoard = [...squares]
 
+    squareToUpdate.value = currentPlayer
+    newBoard[row][column] = squareToUpdate
+    if (currentPlayer.value !== '') {
+      if (currentPlayer === 'X') {
+        setCurrentPlayer(PLAYER_2);
+    } else {
+      setCurrentPlayer(PLAYER_1)
+    };
+    }
+    setSquares(newBoard)
```

Comment:
Instead of calling the checkForWinner function in your return. Call it here inside update squares. You would create a state for the winner then do `setWinner(checkForWinner())`